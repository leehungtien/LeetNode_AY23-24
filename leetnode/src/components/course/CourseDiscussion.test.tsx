import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import CourseDiscussion from './CourseDiscussion';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

const mockedPosts = [
    {
      postId: 1,
      title: 'Sample Post 1',
      // Other fields...
    },
    {
      postId: 2,
      title: 'Sample Post 2',
      // Other fields...
    },
    // Add more mocked post data as needed
];

const mockedCourses = [
    {
        courseName: 'Course 1',
        type: 'Type A',
        courseSlug: 'course-1',
        topics: ['Topic A', 'Topic B'],
        courseLevel: 'Beginner',
        // Other fields...
    },
    {
        courseName: 'Course 2',
        type: 'Type B',
        courseSlug: 'course-2',
        topics: ['Topic C', 'Topic D'],
        courseLevel: 'Intermediate',
        // Other fields...
    },
    // Add more mocked course data as needed
];

const mockedTopics = [
    {
        topicName: 'Topic A',
        // Other fields...
    },
    {
        topicName: 'Topic B',
        // Other fields...
    },
    // Add more mocked topic data as needed
];

// Mock axios to simulate API calls
jest.mock('axios', () => ({
    get: jest.fn((url) => {
      if (url === '/api/forum/getAllPosts') {
        return Promise.resolve({ data: mockedPosts });
      }
      if (url === '/api/forum/getAllCourseNames') {
        return Promise.resolve({ data: mockedCourses });
      }
      if (url === '/api/forum/getAllTopicNames') {
        return Promise.resolve({ data: mockedTopics });
      }
      return Promise.reject(new Error('Not Found'));
    }),
    post: jest.fn(() => Promise.resolve({})),
}));

const queryClient = new QueryClient();

// Mock ResizeObserver
class ResizeObserver {
  observe() {
      // do nothing
  }
  unobserve() {
      // do nothing
  }
  disconnect() {
      // do nothing
  }
}

window.ResizeObserver = ResizeObserver;

describe('CourseDiscussion component', () => {
  it('renders without crashing', () => {
    <QueryClientProvider client={queryClient}>
        <CourseDiscussion courseName="Course 1" />
    </QueryClientProvider>
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <CourseDiscussion courseName="Course 1" />
      </QueryClientProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
