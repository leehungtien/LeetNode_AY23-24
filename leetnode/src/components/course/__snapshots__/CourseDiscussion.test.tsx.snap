// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseDiscussion component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mantine-Stack-root mantine-14a2ivw"
  >
    <div
      class="mantine-Paper-root cursor-pointer transition-shadow duration-300 hover:shadow-lg mantine-ve80bc"
    >
      <div
        class="mantine-6gi1dm"
      >
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1g2btt6"
        />
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Skeleton-root mantine-Skeleton-visible mantine-teg7ze"
      />
      <div
        class="mantine-zaiwzp"
      >
        <div
          class="mantine-879la3"
        >
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1cnh4f3"
          />
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-zb0aoa"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-tk6l3h"
        role="separator"
      />
      <div
        class="mantine-hpmcve"
      >
        <div
          class="mantine-mkrj5s"
        >
          <svg
            class="icon icon-tabler icon-tabler-message"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M4 21v-13a3 3 0 0 1 3 -3h10a3 3 0 0 1 3 3v6a3 3 0 0 1 -3 3h-9l-4 4"
            />
            <line
              x1="8"
              x2="16"
              y1="9"
              y2="9"
            />
            <line
              x1="8"
              x2="14"
              y1="13"
              y2="13"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
          <svg
            class="ml-2"
            fill="none"
            height="24"
            stroke="red"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M19.5 12.572l-7.5 7.428l-7.5 -7.428m0 0a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
    </div>
    <div
      class="mantine-Paper-root cursor-pointer transition-shadow duration-300 hover:shadow-lg mantine-ve80bc"
    >
      <div
        class="mantine-6gi1dm"
      >
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1g2btt6"
        />
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Skeleton-root mantine-Skeleton-visible mantine-teg7ze"
      />
      <div
        class="mantine-zaiwzp"
      >
        <div
          class="mantine-879la3"
        >
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1cnh4f3"
          />
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-zb0aoa"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-tk6l3h"
        role="separator"
      />
      <div
        class="mantine-hpmcve"
      >
        <div
          class="mantine-mkrj5s"
        >
          <svg
            class="icon icon-tabler icon-tabler-message"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M4 21v-13a3 3 0 0 1 3 -3h10a3 3 0 0 1 3 3v6a3 3 0 0 1 -3 3h-9l-4 4"
            />
            <line
              x1="8"
              x2="16"
              y1="9"
              y2="9"
            />
            <line
              x1="8"
              x2="14"
              y1="13"
              y2="13"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
          <svg
            class="ml-2"
            fill="none"
            height="24"
            stroke="red"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M19.5 12.572l-7.5 7.428l-7.5 -7.428m0 0a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
    </div>
    <div
      class="mantine-Paper-root cursor-pointer transition-shadow duration-300 hover:shadow-lg mantine-ve80bc"
    >
      <div
        class="mantine-6gi1dm"
      >
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1g2btt6"
        />
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Skeleton-root mantine-Skeleton-visible mantine-teg7ze"
      />
      <div
        class="mantine-zaiwzp"
      >
        <div
          class="mantine-879la3"
        >
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1cnh4f3"
          />
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-zb0aoa"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-tk6l3h"
        role="separator"
      />
      <div
        class="mantine-hpmcve"
      >
        <div
          class="mantine-mkrj5s"
        >
          <svg
            class="icon icon-tabler icon-tabler-message"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M4 21v-13a3 3 0 0 1 3 -3h10a3 3 0 0 1 3 3v6a3 3 0 0 1 -3 3h-9l-4 4"
            />
            <line
              x1="8"
              x2="16"
              y1="9"
              y2="9"
            />
            <line
              x1="8"
              x2="14"
              y1="13"
              y2="13"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
          <svg
            class="ml-2"
            fill="none"
            height="24"
            stroke="red"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M19.5 12.572l-7.5 7.428l-7.5 -7.428m0 0a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
            />
          </svg>
          <div
            class="mantine-Skeleton-root mantine-Skeleton-visible mantine-1hf0hzn"
          />
        </div>
        <div
          class="mantine-Skeleton-root mantine-Skeleton-visible mantine-8vpxi4"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
